### PR TITLE
feat #8095 - unlimited duration for key

### LINKED ISSUE #8095: Unlimited duration
With the upgrade to v9, PublicLocks can now have unlimited durations! We should support that on the front-end by making sure that when people create their locks they have a way to pick an unlimited duration. For this, let's use a UI similar to the unlimited number of memberships:

Screen Shot 2022-01-27 at 9 05 35 PM
and 
Screen Shot 2022-01-27 at 9 06 01 PM

This change will require updating the UnlockJS library AND the unlock-app dashboard.

### PR DESCRIPTION
**Pull request by kalidiagne**
# Description

This PR adds the possibility to add unlimited duration for a key

# Issues

Fixes #8095 
Refs #
unlimited-1

unlimited-2

# Checklist:

- [ ] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

**Review by julien51 — COMMENTED** [review]
General approach looks good to me!

File: unlock-app/src/components/creator/CreatorLockForm.js [hunk 1]
```diff
@@ -87,12 +89,18 @@ const CreatorLockForm = ({ hideAction, lock, saveLock }) => {
...
-  const handleUnlimitedClick = () => {
+  const handleUnlimitedNumbersOfKeys = () => {
     dispatch({
       change: [{ name: 'maxNumberOfKeys', value: UNLIMITED_KEYS_COUNT }],
     })
   }
 
+  const handleUnlimitedDuration = () => {
+    dispatch({
+      change: [{ name: 'expirationDuration', value: MAX_UINT }],
```
[the remark this answers is not in the document]
Comment (julien51) [reply unlock-app/src/components/creator/CreatorLockForm.js:100]: So please see how `unlockJs` handles unlimited durations and use the right value.

File: packages/unlock-js/src/Unlock/v10/createLock.js [hunk 2]
```diff
@@ -35,7 +35,7 @@ export default async function (lock, callback) {
     maxNumberOfKeys = ETHERS_MAX_UINT
   }
   if (expirationDuration === -1) {
-    maxNumberOfKeys = ETHERS_MAX_UINT
+    expirationDuration = ETHERS_MAX_UINT
```
Comment (kalidiagne) [inline packages/unlock-js/src/Unlock/v10/createLock.js:38]: @julien51 I think it needs to be changed in this way

File: unlock-app/src/components/creator/CreatorLockForm.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (kalidiagne) [reply unlock-app/src/components/creator/CreatorLockForm.js]: @julien51  i used `oneHundredYearsInDays` as value.
The the value used is `unlockJs` is `ethers.BigNumber.from(ethers.constants.MaxUint256).toNumber()`, and is equal to `MAX_UINT` value

**Review by julien51 — APPROVED** [review]
LGTM!

**Pull request merged**